src/hooks/useContracts.test.tsx

```diff
@@ -0,0 +1,75 @@
+import { ERC20 } from '@coordinape/hardhat/dist/typechain';
+import { act, render, waitFor } from '@testing-library/react';
+import { BigNumber } from 'ethers';
+
+import { TestWrapper } from 'utils/testing';
+
+import { useContracts } from './useContracts';
+
+test('return undefined when the web3 provider is not ready', async () => {
+  expect.assertions(1);
+
+  const Harness = () => {
+    const contracts = useContracts();
+    expect(contracts).toBeUndefined();
+    return <></>;
+  };
+
+  await render(
+    <TestWrapper>
+      <Harness />
+    </TestWrapper>
+  );
+});
+
+test('set up contracts', async () => {
+  let balance: BigNumber | undefined;
+
+  const Harness = () => {
```

Comment:
this is my first draft of a decent pattern for tests that depend upon hook &/or component interaction: 
- write a wrapper component (here `Harness` in the sense of "test harness")
- wrap it in `TestWrapper`
- set the `withWeb3` prop to set up the `useWeb3React` hook with a testchain connection
- set variables that were declared in test-local scope, so they can be checked with `await waitFor(...)` later. `waitFor` will retry if anything in its callback throws an exception, until it times out

Reply:
@teeolendo even though i've already merged this, it'd be good for you to take a look at the comment above and this test file to help you write your own frontend<->contract integration tests